src/openrct2/world/Wall.cpp

```diff
@@ -63,35 +63,36 @@ void wall_remove_at(int32_t x, int32_t y, int32_t z0, int32_t z1)
  *
  *  rct2: 0x006E57E6
  */
-void wall_remove_at_z(int32_t x, int32_t y, int32_t z)
+void wall_remove_at_z(const CoordsXYZ& wallPos)
 {
-    wall_remove_at(x, y, z, z + 48);
+    wall_remove_at({ wallPos, wallPos.z, wallPos.z + 48 });
 }
 
 /**
  *
  *  rct2: 0x006E5935
  */
-void wall_remove_intersecting_walls(int32_t x, int32_t y, int32_t z0, int32_t z1, int32_t direction)
+void wall_remove_intersecting_walls(const CoordsXYRangedZ& wallPos, Direction direction)
 {
     TileElement* tileElement;
 
-    tileElement = map_get_first_element_at({ x, y });
+    tileElement = map_get_first_element_at(wallPos);
     if (tileElement == nullptr)
         return;
     do
     {
         if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
             continue;
 
-        if (tileElement->clearance_height <= z0 || tileElement->base_height >= z1)
+        // TODO: It looks like base and clearance are swapped in here
```

Comment:
They are purposely swapped. You are looking for intersections so you compare the base of one item to see if it is below the clearance of another item.

Reply:
Makes sense!